### PR TITLE
Feature 3.16 patient pharmacist consultation scheduling

### PR DESCRIPTION
**Pull request by denisfruza**
https://trello.com/c/zVEeq4P3

### CODE REVIEW

File: backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java [hunk 1]
```diff
@@ -108,7 +114,67 @@ public Boolean cancelAppointment(Long id) {
...
+			pharmacyDTO.setPharmacyId(pharmacy.getId());
+			pharmacyDTO.setPharmacyName(pharmacy.getName());
+			pharmacyDTO.setPharmacyAverageGrade(pharmacy.getAverageGrade());
+			pharmacyDTO.setPharmacyAddress(pharmacy.getAddress().getStreetName() + " " + pharmacy.getAddress().getStreetNumber() + ", " + pharmacy.getAddress().getCity());
+			pharmacyDTO.setLowestPrice(findLowestAppointmentPriceByPharmacyId(pharmacy.getId(), availableAppointments));
+			availablePharmaciesDTOs.add(pharmacyDTO);
+			
+		}
+		return availablePharmaciesDTOs;
+	}
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:134]: Devide this in two methods

File: backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java [hunk 2]
```diff
@@ -108,7 +114,67 @@ public Boolean cancelAppointment(Long id) {
...
+		return lowestPrice;
+	}
+	
+	private Set<Pharmacy> findAvailabePharmaciesByConsultations(Set<Appointment> appointments) {
+		Set<Pharmacy> pharmacies = new HashSet<Pharmacy>();
+		for(Appointment appointment : appointments) {
+			pharmacies.add(appointment.getPharmacy());
+		}
+		return pharmacies;
+	}
```
Comment (dbulaja98) [inline backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:163]: Available pharmacies? :D

Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:163]: yes

Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:163]: That is what they wanted in specification :D

File: backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java (hunk 1 above)
Comment (denisfruza) [reply backend/src/main/java/com/pharmacySystem/service/implementations/AppointmentService.java:134]: fixed

**Review by cakareli — APPROVED** [review]

**Review by dbulaja98 — APPROVED** [review]

**Pull request closed without merging**